src/Instrumentation/ReactHTTP/src/ReactHttpInstrumentation.php

```diff
@@ -0,0 +1,161 @@
+                    ->setAttribute(TraceAttributes::CODE_LINE_NUMBER, $lineno);
+
+                /**
+                 * This cannot be tested with PHPUnit without modifying php.ini with:
+                 * otel.instrumentation.http.request_headers[]="Accept"
+                 */
+                // @codeCoverageIgnoreStart
+                foreach ((array) (get_cfg_var('otel.instrumentation.http.request_headers') ?: []) as $header) {
+                    if ($request->hasHeader($header)) {
+                        $spanBuilder->setAttribute(
+                            sprintf('%s.%s', TraceAttributes::HTTP_REQUEST_HEADER, strtolower($header)),
+                            $request->getHeaderLine($header)
+                        );
+                    }
+                }
+                // @codeCoverageIgnoreEnd
+
+                $span = $spanBuilder->startSpan();
+                $context = $span->storeInContext($parentContext);
+                $propagator->inject($request, HeadersPropagator::instance(), $context);
+
+                Context::storage()->attach($context);
+
+                return [$request];
+            },
+            post: static function (Transaction $transaction, array $params, PromiseInterface $promise, ?Throwable $exception): PromiseInterface {
+                $scope = Context::storage()->scope();
+                $scope?->detach();
+
+                if (!$scope || $scope->context() === Context::getCurrent()) {
+                    return $promise;
+                }
+
+                $span = Span::fromContext($scope->context());
+
+                return $promise->then(
+                    onFulfilled: function (ResponseInterface $response) use ($span) {
+                        $span
+                            ->setAttribute(TraceAttributes::HTTP_RESPONSE_STATUS_CODE, $response->getStatusCode())
+                            ->setAttribute(TraceAttributes::HTTP_RESPONSE_BODY_SIZE, $response->getHeaderLine('Content-Length'));
+
+                        /**
+                         * This cannot be tested with PHPUnit without modifying php.ini with:
+                         * otel.instrumentation.http.response_headers[]="Content-Type"
+                         */
+                        // @codeCoverageIgnoreStart
+                        foreach ((array) (get_cfg_var('otel.instrumentation.http.response_headers') ?: []) as $header) {
+                            if ($response->hasHeader($header)) {
+                                /** @psalm-suppress ArgumentTypeCoercion */
+                                $span->setAttribute(
+                                    sprintf('%s.%s', TraceAttributes::HTTP_RESPONSE_HEADER, strtolower($header)),
+                                    $response->getHeaderLine($header)
+                                );
+                            }
+                        }
+                        // @codeCoverageIgnoreEnd
+
+                        $span->end();
```

Comment:
> Because of the potential for confusion around this, HTTP client library instrumentations SHOULD document their 
behavior around ending HTTP client spans.

The current implementation will lead to different behavior for streaming (span ends after body is fully read) and non-streaming (span ends after response headers are read) responses.

Reply:
I see that now in the specs: https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-client-span-duration

I will think through this, might need to consult @WyriHaximus, as we would have to move the instrumentation farther down the stack, possibly to `React\Http\Io\ClientRequestStream->handleData()` in order to always "end [the HTTP client span] at some point after the HTTP response headers are fully read (or fail to be read)".

Reply:
I have not resolved this yet, just FYI.

Reply:
I added an example and added more to the descriptions of the main ReadMe and the examples ReadMe. Let me know if you feel like this is sufficient to meet these specs.